### PR TITLE
feat: display existing validation checks in publish dialog

### PR DESCRIPTION
**Pull request by jessicamcinchak**
As a first step towards the attached ticket for adding _more_ validation checks on Publish, I'm refactoring a bit how our existing validation checks are organised and displayed.

**Changes:**
- [API] The `/flows/:flowId/diff` endpoint will now always return a list of `validationChecks` when `alteredNodes` is not null (meaning anytime there are changes queued to publish) independent of the status of each validation check
  - Previously, the whole endpoint would return an error for the first failed validation check it found
- [Editor] The "Check for changes to publish" dialog now displays each validation check. If any check has "failed", the "Publish" button will be disabled.
  - Currently, the same validation checks are run across _all flows_, so some checks may appear as "Not applicable" depending on the types of components found in the flow
  - I'm going for a sort of "continuous integration" feel here (green tick dopamine?) - hopefully adding re-assurance to the publishing process
  
Here's how the dialog will display for various validation states (the whole dialog has also been bumped from size "sm" to "md"): 
Screenshot from 2024-06-14 12-51-13

Screenshot from 2024-06-14 12-52-04

Screenshot from 2024-06-14 13-01-10

**Next steps:** 
- Tidy up dialog styling, it's pretty info dense & spacing/margins are definitely a bit misaligned right now
- [Followup PR] Determine which additional validation steps we want to add - see attached ticket for ideas
  - Should all 'failed' validation checks always block ability to Publish or just be a warning? We may want to consider extending the type with another field like `blocking: boolean` that could be used in combination with `status` to determine if Publish button is enabled/disabled?

### CODE REVIEW

**Comment by jessicamcinchak** [comment]
[Validate that flow meets schema requirements on "Publish"](https://trello.com/c/LGIoRgym/2641-validate-that-flow-meets-schema-requirements-on-publish)

**Comment by jessicamcinchak** [comment]
[Validate that flow meets schema requirements on "Publish"](https://trello.com/c/LGIoRgym/2641-validate-that-flow-meets-schema-requirements-on-publish)

File: editor.planx.uk/src/pages/FlowEditor/components/Sidebar/PublishDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+  const getIcon = (status: ValidationCheck["status"]) => {
+    if (status === "Pass") {
+      return <Done color="success" />;
+    } else if (status === "Fail") {
+      return <Close color="error" />;
+    } else if (status === "Not applicable") {
+      return <NotInterested color="disabled" />;
+    }
+  }
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Sidebar/PublishDialog.tsx:243]: nit: An object here might be slightly more readable and typesafe, e.g. 

```ts
const Icon: Record<ValidationCheck["status"], React.FC> = {
  Pass: <Done color="success" />,
  Fail: <Close color="error" />,
  "Not applicable": <NotInterested color="disabled" />
}
```

File: api.planx.uk/modules/flows/validate/validate.test.ts
Comment (DafyddLlyr) [inline api.planx.uk/modules/flows/validate/validate.test.ts:1]: Tests look good! 👍

**Review by DafyddLlyr — APPROVED** [review]
Solid changes! Feels much tidier and more expandable 👍 UI works well for me!

File: editor.planx.uk/src/pages/FlowEditor/components/Sidebar/PublishDialog.tsx (hunk 1 above)
Comment (jessicamcinchak) [reply editor.planx.uk/src/pages/FlowEditor/components/Sidebar/PublishDialog.tsx]: Yes, nice suggestion - updated!

**Review by DafyddLlyr — APPROVED** [review]

**Pull request merged**